nixos/modules/virtualisation/containers.nix

```diff
@@ -124,19 +149,28 @@ in
       };
     };
 
-    environment.etc."containers/containers.conf".source =
-      toml.generate "containers.conf" cfg.containersConf.settings;
-
-    environment.etc."containers/storage.conf".source =
-      toml.generate "storage.conf" cfg.storage.settings;
+    environment.etc = let
+      cdiConfigurationFiles = (lib.attrsets.mapAttrs'
+        (name: value:
+          lib.attrsets.nameValuePair (builtins.concatStringsSep "/" ["cdi" (name + ".json")])
```

Comment:
Lists and other non-stringy types are awesome, but maybe in this case `"cdi/${name}.json"` would've been easier to read